src/Aspire.Hosting.Yarp/api/Aspire.Hosting.Yarp.cs

```diff
@@ -13,11 +13,20 @@ public static partial class YarpServiceExtensions
         public static ApplicationModel.IResourceBuilder<Yarp.YarpResource> AddYarp(this IDistributedApplicationBuilder builder, string name) { throw null; }
 
         public static ApplicationModel.IResourceBuilder<Yarp.YarpResource> WithConfigFile(this ApplicationModel.IResourceBuilder<Yarp.YarpResource> builder, string configFilePath) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<Yarp.YarpResource> WithConfiguration(this ApplicationModel.IResourceBuilder<Yarp.YarpResource> builder, System.Action<Yarp.IYarpConfigurationBuilder> configurationBuilder) { throw null; }
```

Comment:
@benjaminpetit related to my comment:

https://github.com/dotnet/aspire/issues/9878#issuecomment-2974751141

I wonder if instead of having `IYarpConfigurationBuilder` defined the Aspire codebase:

https://github.com/dotnet/aspire/blob/0330128c9b9bf3a2c4e0a877c243a76d0043be47/src/Aspire.Hosting.Yarp/IYarpConfigurationBuilder.cs#L11

I wonder if we should have a configuration builder over in:

https://github.com/dotnet/yarp

That way the type can evolve with YARP itself and then Aspire, and other tools can use it to build configuration files.